server/src/main/java/org/jetbrains/bsp/bazel/server/bep/BepServer.java

```diff
@@ -127,6 +126,39 @@ private void processCompletedEvent(BuildEventStreamProtos.BuildEvent event) {
     }
   }
 
+  private void processActionCompletedEvent(BuildEventStreamProtos.BuildEvent event) {
+    if (event.getId().hasActionCompleted()) {
+      consumeActionCompletedEvent(event);
+    }
+  }
+
+  private void consumeActionCompletedEvent(BuildEventStreamProtos.BuildEvent event) {
+    var label = event.getId().getActionCompleted().getLabel();
+    var actionEvent = event.getAction();
+    if (!actionEvent.getSuccess()) {
+      consumeUnsuccessfulActionCompletedEvent(actionEvent, label);
+    }
+  }
+
+  private void consumeUnsuccessfulActionCompletedEvent(
+      BuildEventStreamProtos.ActionExecuted actionEvent, String label) {
+    String stdErrText = "";
```

Comment:
i meant that instead of changing stdErrText and using it in last two lines, u can extract two last lines to a method and just call it in those ifs